src/aind_behavior_curriculum/trainer.py

```diff
@@ -0,0 +1,393 @@
+"""
+Core Trainer primitive.
+"""
+
+from abc import abstractmethod
+from collections.abc import Iterable
+from typing import List, Optional
+
+from pydantic import Field
+
+from aind_behavior_curriculum import (
+    AindBehaviorModel,
+    Curriculum,
+    Metrics,
+    Policy,
+    Stage,
+    TaskParameters,
+    validate_curriculum,
+)
+
+Stage_Entry = Stage | None
+Policy_Entry = tuple[Policy, ...] | None
```

Comment:
While it doesnt matter for python >3.8 I tend to favor Typing.Tuple over tuple. If using <3.8, tuple is not generic.